activation/activation.go

```diff
@@ -494,13 +488,21 @@ func (b *Builder) Coinbase() types.Address {
 	return b.coinbaseAccount
 }
 
-func (b *Builder) loadChallenge() error {
+func (b *Builder) loadChallenge() (*types.NIPostChallenge, error) {
 	nipost, err := kvstore.GetNIPostChallenge(b.cdb)
 	if err != nil {
-		return err
+		return nil, fmt.Errorf("failed to load nipost challenge from DB: %w", err)
 	}
-	b.challenge = nipost
-	return nil
+	if nipost != nil && nipost.TargetEpoch() < b.currentEpoch() {
+		b.log.With().Info("atx nipost challenge is stale - discarding it",
+			log.FieldNamed("target_epoch", nipost.TargetEpoch()),
+			log.FieldNamed("publish_epoch", nipost.PublishEpoch()),
+			log.FieldNamed("current_epoch", b.currentEpoch()),
+		)
+		b.discardChallenge()
+		return nil, nil
```

Comment:
If we return an `error` here instead of `nil, nil` the code in `PublishActivationTx` could be simplified to:

```golang
        challenge, err := b.loadChallenge()
	if err != nil {
		logger.With().Info("building new atx challenge", log.Stringer("current_epoch", b.currentEpoch()))
		if err := b.buildNIPostChallenge(ctx); err != nil {
		ch, err := b.buildNIPostChallenge(ctx)
		if err != nil {
			return fmt.Errorf("build new atx challenge: %w", err)
		}
		challenge = ch
	}

	logger.With().Info("atx challenge is ready",
		log.Stringer("current_epoch", b.currentEpoch()),
		log.Stringer("publish_epoch", b.challenge.PublishEpoch()),
		log.Stringer("target_epoch", b.challenge.TargetEpoch()))
		log.Stringer("publish_epoch", challenge.PublishEpoch()),
		log.Stringer("target_epoch", challenge.TargetEpoch()),
	)
```

specifically the `nil`-check for `challenge` becomes unnecessary.

Reply:
Done. Thanks for the suggestion!